### DISCUSSION TITLE
Sort data ignoring case

### DISCUSSION THREAD

**Discussion by SShahmuradli** [Blazebit/blaze-persistence#1673] [General]
I am faced with the task of making data sorting case insensitive. Solving a problem using blazebit does not seem like a primitive task. How can I sort by sortBy ignoring the case? Thanks in advance!

public <T> getResultForQuery(
    CriteriaBuilder<T> criteriaBuilder,
    String sortBy,
    boolean asc
) {
    ...
    List<T> resultList = criteriaBuilder.orderBy(sortBy, asc).getResultList();
    ...
}

**Comment by jwgmeligmeyling** [comment]
Case insensitive ordering can be achieved by simply ordering by the lower or uppercase of the string. You'd do this the same in plain JPQL. So `criteriaBuilder.orderBy("UPPER(sortBy)", asc)`.
